sched/definitions.go

```diff
@@ -24,3 +24,26 @@ type Command struct {
 	// We may extend this with more configuration,
 	// E.g. environment variables
 }
+
+// Status for Job & Tasks
+type Status int
+
+const (
+	// NotRunning, waiting to be scheduled
+	NotStarted Status = iota
+
+	// Currently Scheduled and In Progress Job/Task
+	InProgress
+
+	// Successfully Completed Job/Task
+	Completed
+
+	// Job was Aborted, Compensating Tasks are being Applied.
+	// A RollingBack task has not finished its compensating
+	// tasks yet.
+	RollingBack
```

Comment:
As soon as we abort we either know we're 'Completed' (although I don't see the requisite errCode or anything of that nature in JobStatus) or 'RollingBack'. Is that not sufficient, keeping in mind that we already have code that marks something 'Completed' in both the failure/success cases?